cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -1457,30 +1483,44 @@ pTxOut =
                 \Lovelace."
     )
   where
-    parseTxOut :: Atto.Parser (TxOut ShelleyEra)
+    parseTxOut :: Atto.Parser TxOutAnyEra
     parseTxOut =
-      TxOut <$> parseAddressInEra
-            <*  Atto.char '+'
-            <*> (TxOutAdaOnly AdaOnlyInShelleyEra <$> parseLovelace)
+      TxOutAnyEra <$> parseAddressAny
+                  <*  Atto.char '+'
+                  <*> pAdaOnlyValue
 
-pMint :: Parser String
-pMint =
-  Opt.strOption
-    (  Opt.long "mint"
-    <> Opt.metavar "VALUE"
-    <> Opt.help "Mint multi-asset value(s) with the multi-asset cli syntax"
-    <> Opt.value ""
-    )
+pAdaOnlyValue :: Atto.Parser Value
+pAdaOnlyValue = lovelaceToValue <$> parseLovelace
 
-pTxTTL :: Parser SlotNo
-pTxTTL =
+pMintMultiAsset :: Parser Value
+pMintMultiAsset =
+  Opt.option
+    (Opt.eitherReader readValue)
+      (  Opt.long "mint"
+      <> Opt.metavar "VALUE"
+      <> Opt.help "Mint multi-asset value(s) with the multi-asset cli syntax"
+      )
+
+readValue :: String -> Either String Value
+readValue _maCliSyntax = Left "Need 2072 for MA cli syntax parser"
+
+pTxLowerBound :: Parser SlotNo
+pTxLowerBound =
   SlotNo <$>
     Opt.option Opt.auto
-      (  Opt.long "ttl"
+      (  Opt.long "lower-bound"
       <> Opt.metavar "SLOT"
-      <> Opt.help "Time to live (in slots)."
+      <> Opt.help "Time that transaction is valid from (in slots)."
       )
 
+pTxUpperBound :: Parser SlotNo
+pTxUpperBound =
+  SlotNo <$>
+    Opt.option Opt.auto
+      (  Opt.long "upper-bound"
+      <> Opt.metavar "SLOT"
+      <> Opt.help "Time that transaction is valid until (in slots)."
+      )
```

Comment:
Maybe we should also note whether these endpoints (lower and upper bound) are inclusive or exclusive.

Reply:
We should yes. It's inclusive on the lower bound and exclusive on the upper.